### PR TITLE
Add debug endpoint

### LINKED ISSUE #25: add /debug endpoint
we need an endpoint called `/debug` which can return quick information about the server's state and code that its running. an example response should look like

```json
{
  "scetv_githash": "17ae119",
  "scetv_gitbranch": "dev",
  "scetv_gitdirty": 0,
  "state": {
    "process_dict": { },
    "current_video_dict": { }
  },
  "cache": { }
}
```

### game plan
- [x] define `/debug` endpoint that returns a json object
- [x] add a `state` field to the json object that has one field called "process_dict" and another called "current_video_dict" which we can set to the value of both variables. ensure the endpoint returns the value of these dictionaries when called
- [x] once the above is done, add a "cache" field. this should return the verbose value of the `cache` variable. to do this i think we can just do `vars(cache)`? https://www.geeksforgeeks.org/vars-function-python/
- [x] find a command to get the short hash for the latest commit on the branch https://stackoverflow.com/a/949391
- [x] find a command to get the name of the current git branch https://stackoverflow.com/a/6245587
- [x] find a command to check if there are any "stray" uncomitted changes with git. if so, we would set `scetv_gitdirty` to 1. for now just find a command, the 0/1 stuff work with evan on
- [ ] to add the git fields above work with evan on this, we will likely need to set some environment variables before running the server.

### PR DESCRIPTION
**Pull request by weslayer**

### CODE REVIEW

File: modules/cache.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def __str__(self):
         return f"VideoInfo(video_id={self.video_id}, file_path={self.file_path}, size_bytes={self.size_bytes})"
 
 class Cache():
-    def __init__(self, file_path:str, max_size_bytes:int=2_000_000_000) -> None:
+    def __init__(self, file_path:str="not_available", max_size_bytes:int=2_000_000_000) -> None:
```
Comment (evanugarte) [inline modules/cache.py]: do we need this extra param?

File: server.py [hunk 2]
```diff
@@ -210,6 +210,18 @@ def get_metrics():
         media_type="text/plain",
         content=prometheus_client.generate_latest(),
     )
+
+@app.get('/debug')
+def debug():
+    cache = Cache()
```
Comment (evanugarte) [inline server.py]: can we use the global cache variable at the top of this file instead

File: modules/cache.py (hunk 1 above)
Comment (weslayer) [reply modules/cache.py:21]: without it, it gives me: `TypeError: Cache.__init__() missing 1 required positional argument: 'file_path'`

File: server.py (hunk 2 above)
Comment (weslayer) [reply server.py]: ah I did not see that thank you!

**Review by evanugarte — APPROVED** [review]
can you test this on prod before merging

**Comment by weslayer** [comment]
Resolves #26

**Pull request merged**